### PR TITLE
Add illumos target documentation

### PR DESCRIPTION
**Pull request by Noratrieb**
Fixes https://github.com/rust-lang/rust/pull/130132#issuecomment-2339055221

@jclulow @pfmooney I'm adding you as requested.

The page is very barebones (as I do not know illumos well) and could use some improvements (for example in the "Cross-compilation toolchains and C code" section).
Feel free to suggest improvements (or rewrite it from scratch) if you find something.

### CODE REVIEW

File: src/doc/rustc/src/platform-support/illumos.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+The illumos target supports `std` and uses the standard ELF file format.
+
+## Building the target
+
+These targets can be built by adding `aarch64-unknown-illumos` and
+`x86_64-unknown-illumos` as targets in the rustc list.
+
+## Building Rust programs
+
+Rust ships pre-compiled artifacts for the `x86_64-unknown-netbsd` target.
```
Comment (workingjubilee) [inline src/doc/rustc/src/platform-support/illumos.md:26]:
```suggestion
Rust ships pre-compiled artifacts for the `x86_64-unknown-illumos` target.
```

File: src/doc/rustc/src/platform-support/illumos.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Rust does not ship pre-compiled artifacts for `aarch64-unknown-illumos`,
+it requires building the target either as shown above or using `-Zbuild-std`.
+
+## Testing
+
+Tests can be run in the same way as a regular binary.
+
+## Cross-compilation toolchains and C code
+
+The target supports C code.
```
Comment (jclulow) [inline src/doc/rustc/src/platform-support/illumos.md:36]:
```suggestion
The target supports C code.

The illumos project makes available [prebuilt sysroot artefacts](https://github.com/illumos/sysroot) which can be used for cross compilation.  The official Rust binaries are cross-compiled using these artefacts.
```

**Review by jclulow — CHANGES_REQUESTED** [review]
Thanks for getting this started, with my apologies for not having had a chance to do it myself!

**Comment by Noratrieb** [comment]
That took a while, I sort of forgot about this PR, but I finally addressed your feedback and it should be good to go now!

**Comment by jclulow** [comment]
Thanks!

**Review by jclulow — APPROVED** [review]

**Comment by workingjubilee** [comment]
Cool!

@bors r+ rollup

**Comment by tgross35** [comment]
Doesn't this need SUMMARY.md updated and links added?

**Comment by workingjubilee** [comment]
I guess so!

@bors r-

**Comment by Noratrieb** [comment]
💀

**Comment by Noratrieb** [comment]
OK, fixed it and verified that it works locally.
@bors r=workingjubilee

**Pull request merged**